core/CronArchive/QueueConsumer.php

```diff
@@ -201,6 +201,10 @@ public function getNextArchivesToProcess()
                 break;
             }
 
+            if (!$this->isSegmentAvailable($invalidatedArchive)) {
+                continue;
```

Comment:
Ok. So this would mean a segment that doesn't exist anymore or is invalid for other reasons will simply be skipped. But as the record isn't removed it will be checked with each run again. I'm not sure if there would be any benefit in keeping that record, besides the fact, that it would get invalidated as soon as it is valid again. Guess should be safe to remove it. Or am I seeing anything wrong here @tsteur 
```suggestion
                $this->model->deleteInvalidations([$invalidatedArchive]);
                continue;
```